src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -90,6 +97,10 @@ private enum JobHostStartupMode
             HandlingInitializationError = 16 | HandlingNonTransientError
         }
 
+        private bool ShutdownRequested => _applicationStopping == 1 || _applicationStopped == 1;
+
+        private bool NoOtherRestartRequested => Interlocked.Exchange(ref _restartRequested, 1) == 0;
```

Comment:
My issue with this is that the naming suggests that it just checks a value - doesn't imply a modification. Just remove this and inline the Interlocked as you did elsewhere in the method.